src/main/java/org/dcsa/bkg/controller/EventController.java

```diff
@@ -0,0 +1,98 @@
+import org.dcsa.core.extendedrequest.ExtendedRequest;
+import org.dcsa.core.validator.ValidEnum;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.http.server.reactive.ServerHttpResponse;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import javax.validation.ConstraintViolationException;
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Size;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@Validated
+@RequestMapping(
+    value = "events",
+    produces = {MediaType.APPLICATION_JSON_VALUE})
+public class EventController extends AbstractEventController<BKGEventService, Event> {
+
+  private final BKGEventService bkgEventService;
+
+  public EventController(@Qualifier("BKGEventServiceImpl") BKGEventService bkgEventService) {
+    this.bkgEventService = bkgEventService;
+  }
+
+  @Override
+  public BKGEventService getService() {
+    return this.bkgEventService;
+  }
+
+  @Override
+  public String getType() {
+    return "Event";
+  }
+
+  @Override
+  protected ExtendedRequest<Event> newExtendedRequest() {
+    return new ExtendedGenericEventRequest(extendedParameters, r2dbcDialect) {
+      @Override
+      public void parseParameter(Map<String, List<String>> params) {
+        Map<String, List<String>> p = new HashMap<>(params);
+        // Add the eventType parameter (if it is missing) in order to limit the resultset
+        // to *only* SHIPMENT events
+        p.putIfAbsent("eventType", List.of(EventType.SHIPMENT.name()));
+        super.parseParameter(p);
+      }
+    };
+  }
+
+  @GetMapping
+  public Flux<Event> findAll(
+      @RequestParam(value = "shipmentEventTypeCode", required = false)
+          @ValidEnum(clazz = ShipmentEventTypeCode.class)
+          String shipmentEventTypeCode,
+      @RequestParam(value = "documentTypeCode", required = false)
+          @ValidEnum(clazz = DocumentTypeCode.class)
+          String documentTypeCode,
+      @RequestParam(value = "carrierBookingReference", required = false) @Size(max = 35)
+          String carrierBookingReference,
+      @RequestParam(value = "transportDocumentReference", required = false) @Size(max = 20)
+          String transportDocumentReference,
+      @RequestParam(value = "transportDocumentTypeCode", required = false)
+          @ValidEnum(clazz = TransportDocumentTypeCode.class)
+          String transportDocumentTypeCode,
+      @RequestParam(value = "limit", defaultValue = "1", required = false) @Min(1) int limit,
```

Comment:
Have we put 1 as default value in other implementations? I think this should be changed to 20 or 50 - cannot remember what Design Principles say...

Reply:
Just checked application.yaml - here it sets defaultPageSize to 20...

Reply:
Yeah, I can change it, but I also realized that we use this as nothing more than a form of validation. We don't really take these values from here, which maybe we should, we instead take the values directly from ServerHttpRequest. So the default here serves more like documentation.

Reply:
I am not quite sure I follow... IF no limit value is provided - the purpose of the defaultValue is to set it to "1" - meaning only 1 item will be returned if the client doesn't provide a `&limit=xx`?